### PR TITLE
CORS configuration

### PR DESCRIPTION
**Pull request by garrensmith**
CORS configuration in the config section. I've kept it to the simplest settings required for a user to enable and use CORS.

### CODE REVIEW

File: app/addons/cors/components.react.jsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        </div>
+      );
+    }
+
+  });
+
+  var Origins = React.createClass({
+
+    onOriginChange: function (event) {
+      console.log('boom');
```
Comment (robertkowalski) [inline app/addons/cors/components.react.jsx]: debug output

File: app/addons/cors/tests/componentsSpec.react.jsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+      afterEach(function () {
+        React.unmountComponentAtNode(container);
+      });
+
+      /*it('calls change Origin on add radio button change', function () {
+        TestUtils.Simulate.click($(originEl.getDOMNode()).find('input[type="radio"]')[1]);
+        console.log(changeOrigin.calledOnce, $(originEl.getDOMNode()).html());
+        assert.ok(changeOrigin.calledWith(false));
+      });*/
```
Comment (robertkowalski) [inline app/addons/cors/tests/componentsSpec.react.jsx]: can we remove it?

File: app/addons/cors/resources.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  CORS.Config = FauxtonAPI.Model.extend({
+    url: function() {
+      return app.host+"/_config/cors";
+    }
+  });
+
+  CORS.Httpd = FauxtonAPI.Model.extend({
+    url: function() {
+      return app.host+"/_config/httpd";
```
Comment (robertkowalski) [inline app/addons/cors/resources.js]: please put a space between the two and use single quotes

File: app/addons/cors/resources.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  "api"
+],
+
+function (app, FauxtonAPI) {
+  var CORS = FauxtonAPI.addon();
+
+
+  CORS.Config = FauxtonAPI.Model.extend({
+    url: function() {
+      return app.host+"/_config/cors";
```
Comment (robertkowalski) [inline app/addons/cors/resources.js]: please put a space between the two and use single quotes

File: app/addons/cors/templates/origin_domain_table.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations under
+the License.
+-->
+
+<table id="origin-domain-table" class="table table-striped">
+  <tbody></tbody>
+</table>
```
Comment (robertkowalski) [inline app/addons/cors/templates/origin_domain_table.html]: i think we can remove that file

File: app/addons/cors/templates/origin_domain_row.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      Update
+    </a>
+  </div>
+</td>
+<td width="30">
+  <span class="js-edit fonticon-pencil" title="Click to edit"></span>
+</td>
+<td width="30">
+  <span class="js-delete fonticon-trash" title="Click to delete"></span>
+</td>
```
Comment (robertkowalski) [inline app/addons/cors/templates/origin_domain_row.html]: i think we can remove that file

File: app/addons/cors/templates/cors.html [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    
+    </div>
+    
+  </div>
+
+  <div class="form-actions">
+    <input class="btn btn-success" type="submit" value="Save" />
+  </div>
+
+</form>
```
Comment (robertkowalski) [inline app/addons/cors/templates/cors.html]: i think we can remove that file

File: app/addons/cors/views.js [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+      Components.renderCORS(this.el);
+    },
+
+    cleanup: function () {
+      Components.removeCORS(this.el);
+    }
+
+  });
+
+  Views.CORSMain = FauxtonAPI.View.extend({
```
Comment (robertkowalski) [inline app/addons/cors/views.js]: i think we can remove that object

File: app/addons/cors/views.js [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+          });
+        });
+
+        disableCreds.save();
+        disableOrigins.save();
+      }
+    }
+  });
+
+  Views.OriginDomainTable = FauxtonAPI.View.extend({
```
Comment (robertkowalski) [inline app/addons/cors/views.js]: i think we can remove that object

File: app/addons/cors/views.js [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+          model: this.model,
+          index: index
+        }));
+      }, this);
+    }
+  });
+
+
+  // this gets passed the entire model so it can manipulate it directly (add/update the row)
+  Views.OriginDomainRow = FauxtonAPI.View.extend({
```
Comment (robertkowalski) [inline app/addons/cors/views.js]: i think we can remove that object

**Comment by garrensmith** [comment]
Thanks @robertkowalski for the review. I've fixed all the issues you have spotted.

**Comment by robertkowalski** [comment]
after adding an entry & reload the entry is empty: https://cloudup.com/cYlfO2tqejk

**Comment by robertkowalski** [comment]
the input field is overlapping the right side of the screen: 
bildschirmfoto 2015-01-27 um 14 56 22

**Comment by robertkowalski** [comment]
entering the domain `https://foo.com` does not alter the local.ini and does not add the domain

**Comment by garrensmith** [comment]
@robertkowalski you have to enter a domain and press 'Save'.

**Comment by michellephung** [comment]
Hey Garren, 
A. I am also still seeing this issue @robertkowalski mentioned:
screen shot 2015-01-28 at 1 22 23 pm

B. Clicking back and forth between 'Main config' and CORS, will show one of three screens:
- disabled cors (when it actually is enabled, which not right)
- enabled cors, with correct urls (this is right)
- enabled cors, with blank urls (not right)

**Comment by garrensmith** [comment]
Thanks @michellephung and @robertkowalski I've made some fixes. Could you take another look.

**Comment by robertkowalski** [comment]
+1

**Comment by michellephung** [comment]
+1

**Comment by garrensmith** [comment]
Merged

**Pull request closed without merging**